app/webpage/main.js

```diff
@@ -30,46 +37,109 @@ $(document).ready(function () {
-  if(email.val() == "" || email.val() == undefined) {
+  user.email = email.val();
+  var place = user.email.search("@");
+  if(user.email == "" || user.email == undefined || place <= 0) {
+// .search() throws -1 if not found, this makes sure '@' isn't missing or first.
     alert("Please enter your email address.");
-    custTitle(email.val() );
   } else {
-  $(".signin").toggle();
+    user.name = user.email.substring(0,place);
+    custTitle(user.name);
+/* lgin-user-02 */
+  $.ajax({
+    "method": "POST",
+    "crossDomain": true,
+    "url": "http://localhost:6143/api/login/user/",
+    "data": {
+      "email": user.email
+      },
+  /* -01 */
+    "success": function(data) {
+      if(data == "not email")
+        alert("Please enter a valid email address.");
+      else if(data == 1) {
+        $("#username").hide();
+        $("#password").show();
+      } else {
+        $("#username").hide();
+        $("#sign-up").show();
+  
+      }
+    }
+  });
   } 
 }
 
 /* lgin-pswd-01 */
-var pswdValidate = function() {
+var pswdValidate = function(user) {
   var pswd = $("#pswd").val();
   
-  if(pswd != "" && pswd != undefined){
-    $(".top").hide();
-    $("#week-scheduler").show();
-  } else {
+  if(pswd == "" && pswd == undefined){
     alert("Please enter your password.");
+  } else {
+/* lgin-pswd-02 */
+    user.pswd = sjcl.hash.sha256.hash(pswd).toString();
+    
+    $.ajax({
+      "method": "POST",
+      "crossDomain": true,
+      "url": "http://localhost:6143/api/login/pswd/",
+      "data": {
+        "email": user.email,
+        "pswd": user.pswd
+        },
+      "success": function(data) {
+        if(data == 1){
+          $(".top").hide();
+          $("#week-scheduler").show();
+        } else {
+          alert("Password incorrect. Please try again.");
+        }
+      }
+    });
   }
 }
 
 /* lgin-npwd-01 */
-var signupValidate = function() {
+  var signupValidate = function(user) {
   var newPswd = $("#pswd-create").val();
   var check = $("#confirm").val();
   
-    if(newPswd == "") {
-      alert("Please make sure you have entered a password.");
-    } else if(newPswd != check) {
-      alert("Please make sure your passwords match");
-    } else {
-      $(".top").hide();
-      $("#week-scheduler").show();  
-    }
+  if(newPswd == "") {
+    alert("Please make sure you have entered a password.");
+  } else if(newPswd != check) {
+    alert("Please make sure your passwords match");
+  } else {
+    user.pswd = sjcl.hash.sha256.hash(newPswd).toString();
+    
+/* lgin-npwd-02 */
+    $.ajax({
+      "method": "POST",
+      "crossDomain": true,
+      "url": "http://localhost:6143/api/login/create-new/",
+      "data": {
+        "email": user.email,
+        "pswd": user.pswd,
+        "name": user.name
+        },
+      "success": function(data) {
+        if(data.message == "Success") {
+          $(".top").hide();
+          $("#week-scheduler").show();
+        } else {
+          alert(data.message);
```

Comment:
You're doing a lot of alerts to send messages to the client. That's ok for now, but a future revision could be to have a designated user error div somewhere that flashes messages. Alerts are kind of jarring.